tensordict/prototype/tensorclass.py

```diff
@@ -332,6 +334,18 @@ def _batch_size_setter(self, new_size: torch.Size) -> None:
     self.tensordict._batch_size_setter(new_size)
 
 
+def __eq__(self, other):
+    if is_tensorclass(other):
```

Comment:
Should we check that `other` is in fact an instance of the current class, i.e.

```python
if isinstance(other, self.__class__):
    ...
```

Otherwise it's possible that you could have two different dataclasses with the same underlying tensordict that passes this equality check.

Reply:
good point